probability/card_game.cpp

```diff
@@ -0,0 +1,185 @@
+/**
+ * @file
+ * @brief This is a simple card game
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂